tests/sentry/issues/endpoints/test_group_details.py

```diff
@@ -322,24 +322,64 @@ def test_resolve(self):
             user_id=self.user.id, group=group, is_active=True
         ).exists()
 
-    def test_resolved_in_next_release(self):
+    def test_resolved_in_next_release_non_semver(self):
         self.login_as(user=self.user)
 
         project = self.create_project()
         project.flags.has_releases = True
         project.save()
         group = self.create_group(project=project)
         Release.get_or_create(version="abcd", project=project)
+        most_recent_version = Release.get_or_create(version="def", project=project)
+        assert group.status == GroupStatus.UNRESOLVED
+        assert GroupResolution.objects.all().count() == 0
 
         url = f"/api/0/issues/{group.id}/"
-
         response = self.client.put(url, data={"status": "resolvedInNextRelease"})
         assert response.status_code == 200, response.content
 
+        # Refetch from DB to ensure the latest state is fetched
         group = Group.objects.get(id=group.id, project=group.project.id)
         assert group.status == GroupStatus.RESOLVED
 
-        assert GroupResolution.objects.filter(group=group).exists()
+        group_resolution = GroupResolution.objects.filter(group=group).first()
+        assert group_resolution is not None
+        assert group_resolution.group == group
+        assert group_resolution.type == GroupResolution.Type.in_next_release
+        assert group_resolution.status == GroupResolution.Status.pending
+        # This is what should be the result
+        assert group_resolution.release.version == most_recent_version.version
+
+    def test_resolved_in_next_release_semver(self):
+        self.login_as(user=self.user)
+
+        project = self.create_project()
+        project.flags.has_releases = True
+        project.save()
+        Release.get_or_create(version="com.foo.bar@1.0+0", project=project)
+        most_recent_version = Release.get_or_create(version="com.foo.bar@2.0+0", project=project)
+        # It should not be picked up based on creation date
+        Release.get_or_create(version="com.foo.bar@1.0+1", project=project)
```

Comment:
This is the latest created release yet it will not be picked up but the greatest release will be picked up.